src/setup_payload/java/SetupPayloadParser-JNI.cpp

```diff
@@ -40,7 +40,7 @@ jint JNI_OnLoad(JavaVM * jvm, void * reserved)
     return JNI_VERSION_1_6;
 }
 
-JNI_METHOD(jobject, fetchPayloadFromQrCode)(JNIEnv * env, jobject self, jstring qrCodeObj)
+JNI_METHOD(jobject, fetchPayloadFromQrCode)(JNIEnv * env, jobject self, jstring qrCodeObj, jboolean isAllowInvalidPayload)
```

Comment:
isAllowInvalidPayload is a strange name. `acceptInvalidPayloads` or `allowInvalidPayload` or similar sounds better.

What does it mean to allow an invalid payload?

please ad some documentation for this method.

Reply:
or maybe `validatePayload`

Reply:
I changed to allowInvalidPayload and adding some comments.